specs/packagemanager/PackageManagement.md

```diff
@@ -0,0 +1,1136 @@
+* PackageSets -- Batch operations
+* PackageRuntimeManager -- Batch operations for use at runtime via Dynamic Dependencies
+* Usability -- Quality-of-Life enhancements
+
+## 3.1. API Structure
+
+Methods to drive deployment activity typically follow the pattern:
+
+    <verb>Package[Set][By<Type>][Suffix]Async(target, options)
+
+The following verbs are supported:
+
+* Is...Ready
+* Ensure...IsReady
+* Add
+* Stage
+* Register
+* Remove
+* Repair
+* Reset
+* Provision
+* Deprovision
+
+These methods accept their target package(s) as their first parameter. This parameter can be an
+various types of information including:
+
+* Filename
+* Path
+* PackageFamilyName
+* PackageFullName
+* Uri
+
+**NOTE:** Methods with a target of a URI are named ...ByUri....
+
+**NOTE:** Methods with a target of a PackageFamilyName are named ...ByPackageFamilyName....
+
+**NOTE:** Methods with a target of a PackageFamilyName are named ...ByPackageFullName....
+
+**NOTE:** Methods with no ...By<Type>... qualifier accept a string which can contain one or more types
+of identifiers. These can vary for different verbs. See the per-method documentation for the
+specific target types supported by each method.
+
+These methods accept options as a matching `<verb>Package[Set]Options` type, e.g.
+`AddPackageAsync(string packageUri, AddPackageOptions options)`.
+
+The following table shows the supported permutations of verbs and targets:
+
+|Verb         | Path | Filename | PackageFamilyName | PackageFullName | file:  | http(s): | ms-uup: | PackageSet |
+|-------------|:----:|:--------:|:-----------------:|:---------------:|:------:|:--------:|:-------:|:----------:|
+|IsReady      |  X   |    X     |       OS/WAS      |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|EnsureIsReady|  X   |    X     |         X         |        X        |   X    |    X     |  WAS    |    WAS     |
+|Add          | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Stage        | WAS  |    X     |         X         |        X        | OS/WAS |  OS/WAS  | OS/WAS  |    WAS     |
+|Register     | WAS  |  OS/WAS  |       OS/WAS      |      OS/WAS     | OS/WAS |    X     | OS/WAS  |    WAS     |
+|Remove       |  X   |    X     |        WAS        |      OS/WAS     |   X    |    X     | OS/WAS  |    WAS     |
+|Repair       |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Reset        |  X   |    X     |        WAS        |       WAS       |   X    |    X     |  WAS    |    WAS     |
+|Provision    |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+|Deprovision  |  X   |    X     |       OS/WAS      |        X        |   X    |    X     |  WAS    |    WAS     |
+
+Legend:
+
+* OS = Supported by Windows (OS) APIs in the Windows.Management.Deployment.PackageManager namespace.
```

Comment:
By the latest OS version?

Reply:
That's implied. I didn't want to get into that level of detail; microsoft.com docs do it quite well.

See 5.1. Platform Support for more details about functionality and Windows versions.